src/WebSockets/SubscriptionDeterminator.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using GraphQL.Execution;
+using GraphQL.Instrumentation;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+using GraphQL.Validation;
+using GraphQL.Validation.Complexity;
+using GraphQLParser.AST;
+using OperationType = GraphQL.Language.AST.OperationType;
+
+namespace GraphQL.Server.Transports.WebSockets
+{
+    public class SubscriptionDeterminator
+    {
+        private readonly IDocumentBuilder _documentBuilder;
+
+        public SubscriptionDeterminator(): this(new GraphQLDocumentBuilder())
+        {
+        }
+
+        public SubscriptionDeterminator(IDocumentBuilder documentBuilder)
+        {
+            _documentBuilder = documentBuilder;
+        }
+
+        public bool IsSubscription(ExecutionOptions config)
+        {
+
+            config.Schema.FieldNameConverter = config.FieldNameConverter;
+
+            ValidateOptions(config);
+
+            if (!config.Schema.Initialized)
+            {
+                config.Schema.Initialize();
+            }
+
+            var document = config.Document ?? _documentBuilder.Build(config.Query);
```

Comment:
Just a note here, you don't need the `Schema` nor the `ExecutionOptions` to build the `Document`.  All you need is the `Query`.